### PR TITLE
Add Core API tests for Refunds API

### LINKED ISSUE #30675: Write new API test: Refunds > Can refund an order

### PR DESCRIPTION
**Pull request by rodelgc**
### All Submissions:

* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?

### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/woocommerce/issues/30675.

### How to test the changes in this Pull Request:

1. After cloning this branch, setup the e2e environment in your local.
2. Setup the `.env` file using the following values:
    ```
    BASE_URL="http://localhost:8084"
    USER_KEY="admin"
    USER_SECRET="password"
    VERBOSE=true
    ```
3. Run the tests for the Refunds API in this PR: `npm test -- --group=refunds`
4. All refunds tests should pass:
    Screen Shot 2021-10-19 at 8 34 12 PM 1
5. Login to WP Admin and navigate to WooCommerce > Orders. Verify that there are no orders in the list. This means that the tests correctly cleaned up the order it created.
6. Go to Products page. Verify that there are no products listed. This means that the test had correctly cleaned up the product it created.

### Changelog entry

> Added API tests for the Refunds API.

### FOR PR REVIEWER ONLY:

* [x] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Comment by zhongruige** [comment]
@rodelgc It looks like this PR also contains changes for the coupons tests (example: https://github.com/woocommerce/woocommerce/pull/30982/files#diff-8973c6c70700619c0c3857fd8cf0ac48fe12e73ea08ce54fa665158ccdfc8c8eR1). Can you merge in the latest `trunk` so this just has the refunds changes? Thanks!

**Comment by rodelgc** [comment]
Hi @zhongruige, ah right, I think I based off this branch from the coupons tests instead of `trunk`. Done updating this branch with `trunk` now, and ready for review.

File: plugins/woocommerce/tests/e2e/api-core-tests/tests/refunds/refunds.test.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
+const { refundsApi } = require( '../../endpoints/refunds' );
+const { ordersApi } = require( '../../endpoints/orders' );
+const { productsApi } = require( '../../endpoints/products' );
+const { refund } = require( '../../data' );
+const { expect } = require( '@jest/globals' );
```
Comment (zhongruige) [inline plugins/woocommerce/tests/e2e/api-core-tests/tests/refunds/refunds.test.js]: We can drop this import, the `@jest/globals` (specifically `expect` here) are available in the test run.

File: plugins/woocommerce/tests/e2e/api-core-tests/CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,7 @@
 # Unreleased
 
 - Added API tests for the Coupons API.
+- Added API tests for the Refunds API.
```
Comment (zhongruige) [inline plugins/woocommerce/tests/e2e/api-core-tests/CHANGELOG.md]: Just noticing this, we can actually move these two lines (adding the coupons and refunds API tests) into the `0.1.0` version block before as this package hasn't been released yet.

**Review by zhongruige — COMMENTED** [review]
Thanks for working on this @rodelgc!

Just two minor comments on the CHANGELOG entry and removing the `@jest/globals` import from the test. After reviewing those we should be good to merge.

File: plugins/woocommerce/tests/e2e/api-core-tests/tests/refunds/refunds.test.js (hunk 1 above)
Comment (rodelgc) [reply plugins/woocommerce/tests/e2e/api-core-tests/tests/refunds/refunds.test.js]: Thanks @zhongruige! This must be my IDE doing funny things. Done removing this import in 919f41fed3e2d908f4f376150a16cc1d02b35eb1.

File: plugins/woocommerce/tests/e2e/api-core-tests/CHANGELOG.md (hunk 2 above)
Comment (rodelgc) [reply plugins/woocommerce/tests/e2e/api-core-tests/CHANGELOG.md]: Ah right, makes sense! Thanks! I have now moved them to the `0.1.0` block and added the "Added" subheader to be more consistent with the changelog format in e2e. Let me know if that's fine.

File: plugins/woocommerce/tests/e2e/api-core-tests/tests/refunds/refunds.test.js (hunk 1 above)
Comment (zhongruige) [reply plugins/woocommerce/tests/e2e/api-core-tests/tests/refunds/refunds.test.js]: I find I need to wrangle my IDE in as well, it tends to be overly helpful with imports! Thanks for removing this line!

File: plugins/woocommerce/tests/e2e/api-core-tests/CHANGELOG.md (hunk 2 above)
Comment (zhongruige) [reply plugins/woocommerce/tests/e2e/api-core-tests/CHANGELOG.md]: Looks good!

**Review by zhongruige — APPROVED** [review]
Thanks for adding these in @rodelgc! These ran great for me locally:

```
 PASS  tests/refunds/refunds.test.js (7.123s)
  Refunds API tests
    ✓ can create a refund (1129ms)
    ✓ can retrieve a refund (644ms)
    ✓ can list all refunds (505ms)
    ✓ can delete a refund (1654ms)

Test Suites: 1 passed, 1 of 5 total
Tests:       4 passed, 4 total
Snapshots:   0 total
Time:        8.194s, estimated 38s
Ran all test suites.
```

**Pull request merged**